opentracing-util/src/main/java/io/opentracing/util/AutoFinishScopeManager.java

```diff
@@ -41,4 +41,9 @@ public Span activeSpan() {
         AutoFinishScope scope = tlsScope.get();
         return scope == null ? null : scope.span();
     }
+
+    @Override
+    public void clear() {
+        tlsScope.set(null);
```

Comment:
Shouldn't this be `tlsScope.remove()` ?

Reply:
I think both have the same semantics but `tlsScope.remove()` removes the entry in the thread local map whereas `set(null)` still keeps it, which leads to fewer allocations of `java.lang.ThreadLocal.ThreadLocalMap.Entry` objects.

Reply:
@felixbarny sounds ok.
@carlosalberto  If this is the reasoning, could you please add an (inline) comment so future developers also know this?

Reply:
@sjoerdtalsma Hey, added a small note - does it look fine to you?

Reply:
Yes, as this would have stopped me from asking 'why not remove()' :wink: